static/app/components/events/interfaces/performance/utils.tsx

```diff
@@ -48,3 +54,44 @@ export function getSpanInfoFromTransactionEvent(
     causeSpans: causeSpanIDs.map(spanID => spansById[spanID]),
   };
 }
+
+/**
+ * Given an event for a performance issue, returns the `affectedSpanIds` and `focusedSpanIds`.
+ * Both of these subsets of spans are used to determine which spans are initially visible on the span tree on the issue details
+ * page. The main difference is that the former will be highlighted in red, these spans are intended to indicate the 'root cause' spans
+ * of the issue, with the latter being supplemental spans that are involved in the issue but not necessarily the cause of it.
+ *
+ * @param event
+ */
+export function getProblemSpansForSpanTree(event: EventTransaction): {
+  affectedSpanIds: string[];
+  focusedSpanIds: string[];
+} {
+  const issueType = event?.perfProblem?.issueType;
+  const affectedSpanIds: string[] = [];
+  const focusedSpanIds: string[] = [];
+
+  // By default, offender spans will always be `affected spans`
+  const offenderSpanIds = event.perfProblem?.offenderSpanIds ?? [];
+  affectedSpanIds.push(...offenderSpanIds);
+
+  if (issueType !== IssueType.PERFORMANCE_N_PLUS_ONE_API_CALLS) {
+    const parentSpanIds = event.perfProblem?.parentSpanIds ?? [];
+    affectedSpanIds.push(...parentSpanIds);
+  }
+
+  if (issueType === IssueType.PERFORMANCE_CONSECUTIVE_DB_QUERIES) {
+    const consecutiveSpanIds = event?.perfProblem?.causeSpanIds ?? [];
+
+    if (consecutiveSpanIds.length < 11) {
+      focusedSpanIds.push(...consecutiveSpanIds);
+    }
+  }
+
+  if (issueType === IssueType.PERFORMANCE_N_PLUS_ONE_DB_QUERIES) {
+    const precedingSpans = event.perfProblem?.causeSpanIds ?? [];
+    focusedSpanIds.push(...precedingSpans);
+  }
+
+  return {affectedSpanIds, focusedSpanIds};
+}
```

Comment:
The inconsistent use of `event?` vs `event` throws me a bit here - the nullability of event shouldn't differ by branch right?

Also, the number of `IssueType`-specific logic here makes me a touch suspicious that maybe this logic should also be split by type rather than colocated in one method. Just something to think about as the number of detector types grows - it's fine as is.

Those are both minor and LGTM regardless. I'd be very happy to see this change merged 😁

Reply:
Good eye! The inconsistent optional chaining on `event` seems to be an artifact of a previous time, I'll clean those up.

You bring up a good point about separating the logic by type, I'll certainly keep an eye on it and refactor if needed in the near future